kernel/safe_typing.ml

```diff
@@ -883,30 +885,39 @@ let export_private_constants eff senv =
   let senv = List.fold_left add_constant_aux senv bodies in
   exported, senv
 
+  let opacify_body senv cb =
+    let c = match cb.const_body with
+      | Def x -> x
+      | _ -> assert false in
+    let fc = Future.from_val (Mod_subst.force_constr c, Opaqueproof.PrivateMonomorphic Univ.ContextSet.empty) in (* TODO *)
```

Comment:
I kind of managed to make it type, but I don't know what I should do here w.r.t. the universe part.